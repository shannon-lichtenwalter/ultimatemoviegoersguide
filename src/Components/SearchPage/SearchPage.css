form {
  margin: 10px;
  border: none;
  font-size: 1.5rem; 
  background-color: transparent;
}

fieldset {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: rgb(231, 231, 227, 0.4);
  border: none;
  border-radius: 5px;
  margin-bottom: 5%;
}

label {
  text-align: center;
}

input {
  width: 80%;
  margin: 10px auto;
  font-size: 1.5rem;
  color: grey;
  border-radius: 5px;
  border: 1px solid lightgrey;
}