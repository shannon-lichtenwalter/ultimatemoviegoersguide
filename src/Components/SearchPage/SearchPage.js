import React from 'react';
import GetMovieLists from '../../Services/getMovieLists';
import Movie from '../Movie/Movie';

class SearchPage extends React.Component {
  state = {
    searchField: null,
    data: null
  }

  handleSearch = (e) => {
    this.setState({ 
      searchField: e.target.value 
    }, this.searchMovies())
  }

  searchMovies = () => {
    console.log(this.state.searchField)
    if (this.state.searchField) {
      GetMovieLists.searchFilms(this.state.searchField)
      .then((res) => {
        this.setState({
          data:res
        })
      })
      .catch(e => this.props.setError(e));
    }
  }

  componentDidMount = () => {
    if (this.props.error) {
      this.props.resetError()
    }
  }

  render() {
    return (
      <section>
        <form onSubmit={(e)=>e.preventDefault()}>
          <fieldset>
            <label htmlFor='search'>Search For A Movie</label>
            <input type='text' id='search' name='search' onChange={(e) => this.handleSearch(e)}></input>
          </fieldset>
        </form>
        <ul>
          {this.state.data && this.state.data.results.map((movie, index) => {
            return <Movie key={index} data={movie} showDetails={this.props.showDetails}/>
          })}
        </ul>
      </section>
    )
  }
}

export default SearchPage;