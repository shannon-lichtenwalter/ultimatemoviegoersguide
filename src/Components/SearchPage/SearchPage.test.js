import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchPage from './SearchPage';

describe('SearchPage Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render
      (<BrowserRouter><SearchPage /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  }); 

  it('Displays the SearchPage component correctly when rendered', () => {
    const wrapper = mount(<BrowserRouter><SearchPage /></BrowserRouter>);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
});