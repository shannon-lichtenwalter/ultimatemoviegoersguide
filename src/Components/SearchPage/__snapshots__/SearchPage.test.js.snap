// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage Component Displays the SearchPage component correctly when rendered 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SearchPage>
      <section>
        <form
          onSubmit={[Function]}
        >
          <fieldset>
            <label
              htmlFor="search"
            >
              Search For A Movie:
            </label>
            <input
              id="search"
              name="search"
              onChange={[Function]}
              type="text"
            />
          </fieldset>
        </form>
        <ul />
      </section>
    </SearchPage>
  </Router>
</BrowserRouter>
`;
